virttest/qemu_devices/qdevices.py

```diff
@@ -2091,11 +2092,28 @@ def _dev2addr(self, device):
                     bus = int(busid[1])
         if isinstance(busid, int):
             bus = busid
-        if device.get_param('unit'):
-            unit = int(device.get_param('unit'))
+        if device.get_param(self.unit_spec):
+            unit = int(device.get_param(self.unit_spec))
         return [bus, unit]
 
 
+class QSerialBus(QBusUnitBus):
+
+    """ Serial bus representation """
+
+    def __init__(self, busid, bus_type, aobject=None, max_ports=32):
+        """
+        :param busid: bus id
+        :param bus_type: bus type(virtio-serial-device, virtio-serial-pci)
+        :param aobject: autotest object
+        :param max_ports: max ports, default 32
+        """
+
+        super(QSerialBus, self).__init__(busid, 'SERIAL', [1, max_ports],
+                                         aobject, bus_type, unit_spec='nr')
```

Comment:
Could you explain which device has option `nr` ?   I can't get it from qemu side.

Reply:
@yanan-fu 
```
#  /usr/libexec/qemu-kvm -device virtserialport,?
virtserialport.name=str
virtserialport.chardev=str (ID of a chardev to use as a backend)
virtserialport.nr=uint32
```
That should the virtio-serial bus distributing to each device.